makefile

```diff
@@ -1,8 +1,36 @@
+DMD_ROOT_SRC := \
+	$(shell find dmd/compiler/src/dmd/common -name "*.d")\
+	$(shell find dmd/compiler/src/dmd/root -name "*.d")
+DMD_LEXER_SRC := \
+	dmd/compiler/src/dmd/console.d \
+	dmd/compiler/src/dmd/entity.d \
+	dmd/compiler/src/dmd/errors.d \
+	dmd/compiler/src/dmd/errorsink.d \
+	dmd/compiler/src/dmd/location.d \
+	dmd/compiler/src/dmd/file_manager.d \
+	dmd/compiler/src/dmd/globals.d \
+	dmd/compiler/src/dmd/id.d \
+	dmd/compiler/src/dmd/identifier.d \
+	dmd/compiler/src/dmd/lexer.d \
+	dmd/compiler/src/dmd/tokens.d \
+	dmd/compiler/src/dmd/utils.d \
+	$(DMD_ROOT_SRC)
+
+DMD_PARSER_SRC := \
+	dmd/compiler/src/dmd/astbase.d \
+	dmd/compiler/src/dmd/parse.d \
+	dmd/compiler/src/dmd/parsetimevisitor.d \
+	dmd/compiler/src/dmd/transitivevisitor.d \
+	dmd/compiler/src/dmd/permissivevisitor.d \
+	dmd/compiler/src/dmd/strictvisitor.d \
+	dmd/compiler/src/dmd/astenums.d \
+	$(DMD_LEXER_SRC)
```

Comment:
Might `dmd -i` be useful here? This makes you quite brittle towards upstream changes.

Reply:
Yeah I'm not really sure about keeping these changes, because they refused to work once I started using `ASTCodegen`. I tried a bunch of things to fix it, but didn't manage to make it work. dmdfmt currently works with `dub build`, but the Makefile itself is pretty broken. Even with every single D file in DMD included, it would die during linking.